### ISSUE TITLE
Enable Caravel for secured repositories

### ISSUE THREAD

**Issue by kazzkiq** [caravel-tool/caravel#4]
Repositories using SSH keys or passwords are not supported yet _(and thats bad)_.

**Password:**
These repos end up prompting for user password. But as Node runs the `git clone` command indirectly (through Caravel), the prompt never shows to the user. No password typed, Caravel can't carry on. Caravel uses [commander](https://www.npmjs.com/package/commander) for CLI so perhaps we would need to dig a little more on its API.

**SSH keys:**
To be honest I don't know how this work and have no experience with repos using SSH keys. It seems that it could be solved outside Caravel (User must register a key on their local files and it gets activated automatically when needed). But I'm really not sure. Perhaps, if this isn't an issue for the project itself, we could still add a section to the website explaining how to do it.

---
## TODO:
- [ ] Enable Caravel for Passwords
- [x] Enable Caravel for SSH

**Comment by kazzkiq** [comment]
On the Password issue:
It seems Node support CLI prompt writing with `.stdin()` function. E.g.

``` js
const exec = require('child_process').exec;

let cmd = exec('git command here etc', (error, stdout, stderr) => {
    // ...
});

// When a prompt appears, write "Yes" (could be the password)
cmd.stdin.write("yes");
```

More info **[here](http://stackoverflow.com/questions/25694884/automatic-writing-text-to-the-console-using-node-js/25695188)**.

**Comment by felipeuntill** [comment]
I good approach to solve both problems is use the github [access token for command line](https://help.github.com/articles/creating-an-access-token-for-command-line-use/)

Can i implement it?

**Comment by kazzkiq** [comment]
@felipeuntill Actually I think an universal approach would be better, since users can then be able to use other Git services too ([Gitlab](https://about.gitlab.com/), [Bitbucket](https://bitbucket.org/), [Assembla](https://www.assembla.com/), [Beanstalk](http://beanstalkapp.com/), etc).

I think the best approach would be dropping password support (this approach requires Git password on the `caravel.json` file, which is dangerous and not secure practice) and using SSH login instead.

The SSH don't need to be configured by Caravel, but we can still add a tutorial on the future documentation/website so users know how to do it on their systems.

This answer explains about how to configure Git SSH authentication:
http://stackoverflow.com/a/11251797/694105

Theoretically those are the needed steps:
1. User register SSH key for their repo (on their machine);
2. User install and run Caravel;
3. Caravel emit Git commands, which automatically use the registered SSH from the system;
4. Everything works without need for passwords. 

The problem is, this is theory. I haven't tested this approach yet. But as most Git services have tutorials on how to enable SSH for repos, I believe it will work as expected.

---

**tl;dr:**
We should drop password support and accept only SSH-based access for private repos (which need to be done in the system by the user, not by Caravel).

So, the next step now would be removing password support on Caravel and testing SSH keys instead.

More information about this approach here:
https://help.github.com/articles/generating-an-ssh-key/
http://docs.gitlab.com/ce/ssh/README.html
https://git-scm.com/book/pt-br/v1/Git-no-Servidor-Gerando-Sua-Chave-P%C3%BAblica-SSH

**Comment by kazzkiq** [comment]
Just tested with a private project, and after adding SSH keys it work flawlessly. Will be closing this one.
